mitodl/ocw-studio
# Multiple open catalog webhook endpoints

### What are the relevant tickets?
Closes #1994

Related salt-ops PR: https://github.com/mitodl/salt-ops/pull/1609

### Description (What does it do?)
Modifies the pipeline generation code to allow for sending webhook requests to multiple urls.
Renames the following env vars:
  - `OCW_NEXT_SEARCH_WEBHOOK_KEY` to `OPEN_CATALOG_WEBHOOK_KEY`
  -  `OPEN_DISCUSSIONS_URL` to `OPEN_CATALOG_URLS`

### How can this be tested?
- Change the value of `OCW_STUDIO_ENVIRONMENT` to anything except `dev`
- Add this to your .env: 
  ```
  OPEN_CATALOG_URLS=http://discussions-rc.odl.mit.edu,https://mitopen-rc.odl.mit.edu
  OPEN_CATALOG_WEBHOOK_KEY=<former value of OCW_NEXT_SEARCH_WEBHOOK_KEY>
  ```
- Run the following commands:
  ```
  ./manage.py upsert_theme_assets_pipeline --delete
  ./manage.py upsert_mass_build_pipeline --delete
  ./manage.py upsert_site_removal_pipeline --delete
  ./manage.py backpopulate_pipelines --filter <site-name>
  ```
- Using fly, save the pipeline configurations locally and make sure there are sections present to send webhooks to both open catalog urls:
  ```
   fly -t local get-pipeline -p 'mass-build-sites/offline:false,prefix:"",projects_branch:main,starter:"",themes_branch:main,version:live ' > mass_publish.yml
   fly -t local get-pipeline -p remove-unpublished-sites > remove_unpublished.yml
   fly -t local get-pipeline -p live/site:site_name > single_site.yml
  ```

### Additional Context
This assumes that the webhook key is always the same for all endpoints.  In salt-ops they all use the [same source](https://github.com/mitodl/salt-ops/blob/main/pillar/heroku/mitopen.sls#L127).

When deployed to RC/production, all the existing pipelines will need to be updated
